### DISCUSSION TITLE
Minimum RPC API support

### DISCUSSION THREAD

**Discussion by mattludwigs** [ExFacto/oraclex#6] [General]
The RPC calls that I am tracking that are necessary for this project are:

- getblockheight
- getblockinfo
- Sendrawtransaction
- getmempool(stats)
- adddescriptortowallet
- addmultisigaddress
- estimatesmartfee
- gettxout

Are there others I am missing? 

cc @SachinMeier

**Comment by SachinMeier** [comment]
I'm leaning towards having the Client app just use bitcoin core's wallet, at least at first, since building wallet funcitonality in will be a large amount more lift. In that case we'd need several wallet-related RPC calls. I don't even have that planned out yet. Will update here once that is planned

**Comment by SachinMeier** [comment]
Update: Since we need the signer to create Adaptor Signatures, we can't use Bitcoind for the actual CET or Refund Tx signing. And, if we ever want to use a Barrier Oracle, we'd need to use Adaptor Signatures for the funding transaction as well. 

Thus, we will need to build our own internal wallet. 

With this in mind, the tentative list of RPC calls needed are: 
- getblockheight
- getblockinfo
- sendrawtransaction
- getrawtransaction (with verbose & blockhash)
- getmempool(stats)
- adddescriptortowallet
- addmultisigaddress
- estimatesmartfee
- gettxout
- importaddress
- importprivkey

Nice to Haves: 
- signmessagewithprivkey
- validateaddress
- verifymessage

Later, once we add PSBT support (not MVP):
- finalizepsbt
- utxoupdatepsbt
- signrawtransactionwithkey
- joinpsbts
- createpsbt
- converttopsbt
- combinepsbt

**Reply by mattludwigs** [reply]
I will add this as the roadmap for `btcrpc`.
